src/rules/noObjectLiteralTypeAssertionRule.ts

```diff
@@ -86,6 +86,13 @@ function walk(ctx: Lint.WalkContext<Options>): void {
         if (
             isAssertionExpression(node) &&
             node.type.kind !== ts.SyntaxKind.AnyKeyword &&
+            // Allow const assertions, introduced in TS 3.4
+            !(
+                node.type.kind === ts.SyntaxKind.TypeReference &&
+                (ts.isConstTypeReference !== undefined
+                    ? ts.isConstTypeReference(node.type)
+                    : node.type.getText(ctx.sourceFile) === "const")
```

Comment:
> `node.type.getText(ctx.sourceFile) === "const"`

If the user isn't running TSLint with a TypeScript version that understands `as const` then TypeScript will already be giving them errors. You can ignore this case, I think.

Reply:
Agreed.